LiteCore/RevTrees/RawRevTree.cc

```diff
@@ -45,7 +46,7 @@ namespace litecore {
 
     std::deque<Rev> RawRevision::decodeTree(slice raw_tree, RevTree::RemoteRevMap& remoteMap,
                                             std::vector<const Rev*>& rejectedRevs, RevTree* owner, sequence_t curSeq) {
-        const RawRevision* rawRev = (const RawRevision*)raw_tree.buf;
+        auto rawRev = (const RawRevision*)raw_tree.buf;
```

Comment:
const auto ?  I don't actually know what rules apply to auto when casting explicitly to a const, but `const auto` is a thing.

Reply:
The variable still has type `const RawRevision*`. I think using `const auto` can add const to a value you don't want to change, but using `auto` for a type that is already const can't take the const-ness away.